code/modules/xenoarchaeology/traits/xenoartifact_minors.dm

```diff
@@ -0,0 +1,430 @@
+	man.forceMove(X) //Better hope no greedy goblins took all the zlevels
+	man.anchored = TRUE
+	var/obj/effect/proc_holder/spell/targeted/xeno_senitent_action/P = new /obj/effect/proc_holder/spell/targeted/xeno_senitent_action(,X)
+	man.AddSpell(P)
+	if(man.key)
+		playsound(get_turf(X), 'sound/items/haunted/ghostitemattack.ogg', 50, TRUE)
+	qdel(S)
+
+/obj/effect/proc_holder/spell/targeted/xeno_senitent_action //Lets sentience target goober
+	name = "Activate"
+	desc = "Select a target to activate your traits on."
+	range = 1
+	charge_max = 0 SECONDS
+	clothes_req = 0
+	include_user = 0
+	action_icon = 'icons/mob/actions/actions_revenant.dmi'
+	action_icon_state = "r_transmit"
+	action_background_icon_state = "bg_spell"
+	var/obj/item/xenoartifact/xeno
+
+/obj/effect/proc_holder/spell/targeted/xeno_senitent_action/Initialize(mapload, var/obj/item/xenoartifact/Z)
+	. = ..()
+	xeno = Z
+	range = Z.max_range+3
+
+/obj/effect/proc_holder/spell/targeted/xeno_senitent_action/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	for(var/atom/M in targets)
+		if(xeno)
+			xeno.true_target = list(M)
+			xeno.default_activate(xeno.charge_req+50)
+			charge_max = xeno.cooldown+xeno.cooldownmod
+
+/datum/xenoartifact_trait/minor/sentient/Destroy(force, ...)
+	. = ..()
+	QDEL_NULL(man) //Kill the inner person. Otherwise invisible runs around
+	QDEL_NULL(S)
+
+/obj/effect/mob_spawn/sentient_artifact
+	death = FALSE
+	name = "Sentient Xenoartifact"
+	short_desc = "You're a maleviolent sentience, possesing an ancient alien artifact."
+	flavour_text = "Return to your master..."
+	use_cooldown = TRUE
+	invisibility = 101
+	var/obj/item/xenoartifact/artifact
+
+/obj/effect/mob_spawn/sentient_artifact/Initialize(mapload, var/obj/item/xenoartifact/Z)
+	if(!Z)
+		qdel(src)
+		return FALSE
+	artifact = Z
+	return ..()
+
+/obj/effect/mob_spawn/sentient_artifact/create(ckey, name)
+	var/datum/xenoartifact_trait/minor/sentient/S = artifact.get_trait(/datum/xenoartifact_trait/minor/sentient)
+	S.setup_sentience(artifact, ckey)
+
+//============
+// Delicate, makes the artifact have limited uses
+//============
+/datum/xenoartifact_trait/minor/delicate
+	desc = "Fragile"
+	label_desc = "Fragile: The Artifact is poorly made. Continuous use will destroy it."
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/delicate/on_init(obj/item/xenoartifact/X)
+	X.max_integrity = pick(200, 300, 500, 800, 1000)
+	X.obj_integrity = X.max_integrity
+	X.alpha = X.alpha * 0.55
+
+/datum/xenoartifact_trait/minor/delicate/activate(obj/item/xenoartifact/X, atom/user)
+	if(X.obj_integrity)
+		X.obj_integrity -= 100
+	else if(X.obj_integrity <= 0)
+		X.visible_message("<span class='danger'>The [X.name] shatters!</span>", "<span class='danger'>The [X.name] shatters!</span>")
+		var/obj/effect/decal/cleanable/ash/A = new(get_turf(X))
+		A.color = X.material
+		playsound(get_turf(X), 'sound/effects/glassbr1.ogg', 50, TRUE) 
+		qdel(X)
```

Comment:
There should be a way to reasonably tell that this is fragile without code diving. Each use should make a noise or have a visible_message